pkg/host_cleaning/host_purge.go

```diff
@@ -30,95 +25,32 @@ func HostPurge(ctx context.Context, containerBackend container_backend.Container
 	}
 
 	if err := logboek.Context(ctx).LogProcess("Running werf docker containers purge").DoError(func() error {
-		if err := werfContainersFlushByFilterSet(ctx, filters.NewArgs(), commonOptions); err != nil {
-			return err
-		}
-
-		return nil
+		return purger.FlushContainers(ctx, commonOptions)
 	}); err != nil {
 		return err
 	}
 
 	if err := logboek.Context(ctx).LogProcess("Running werf docker images purge").DoError(func() error {
-		filterSet := filters.NewArgs()
-		filterSet.Add("label", image.WerfLabel)
-
-		if err := werfImagesFlushByFilterSet(ctx, filterSet, commonOptions); err != nil {
-			return err
-		}
-
-		localManagedImageRecordImageNameFormat := "werf-managed-images/%s" // legacy
-		filterSet = filters.NewArgs()
-		filterSet.Add("reference", fmt.Sprintf(localManagedImageRecordImageNameFormat, "*"))
-
-		if err := werfImagesFlushByFilterSet(ctx, filterSet, commonOptions); err != nil {
-			return err
-		}
-
-		localMetadataImageRecordImageNameFormat := "werf-images-metadata-by-commit/%s" // legacy
-		filterSet = filters.NewArgs()
-		filterSet.Add("reference", fmt.Sprintf(localMetadataImageRecordImageNameFormat, "*"))
-
-		if err := werfImagesFlushByFilterSet(ctx, filterSet, commonOptions); err != nil {
-			return err
-		}
-
-		return nil
+		return purger.FlushImages(ctx, commonOptions)
 	}); err != nil {
 		return err
 	}
 
-	if err := tmp_manager.Purge(ctx, commonOptions.DryRun, containerBackend); err != nil {
+	if err := tmp_manager.Purge(ctx, commonOptions.DryRun, backend); err != nil {
 		return fmt.Errorf("tmp files purge failed: %w", err)
 	}
 
 	if err := logboek.Context(ctx).LogProcess("Running werf home data purge").DoError(func() error {
-		return purgeHomeWerfFiles(ctx, commonOptions.DryRun, containerBackend)
+		return purger.PurgeWerfHomeFiles(ctx, commonOptions)
 	}); err != nil {
 		return err
 	}
 
 	if err := logboek.Context(ctx).LogProcess("Deleting stapel").DoError(func() error {
```

Comment:
Here is a backend-specific case. For the buildah backend, there is nothing to delete.